libs/platform/ebpf_tracelog.h

```diff
@@ -0,0 +1,405 @@
+        _In_ const ebpf_utf8_string_t* string);
+#define EBPF_LOG_MESSAGE_UTF8_STRING(trace_level, keyword, message, value)            \
+    if (TraceLoggingProviderEnabled(ebpf_tracelog_provider, trace_level, keyword)) {  \
+        ebpf_log_message_utf8_string(_##trace_level##, _##keyword##, message, value); \
+    }
+
+    void
+    ebpf_log_message_ntstatus(
+        ebpf_tracelog_level_t trace_level,
+        ebpf_tracelog_keyword_t keyword,
+        _In_z_ const char* message,
+        NTSTATUS status);
+#define EBPF_LOG_MESSAGE_NTSTATUS(trace_level, keyword, message, status)             \
+    if (TraceLoggingProviderEnabled(ebpf_tracelog_provider, trace_level, keyword)) { \
+        ebpf_log_message_ntstatus(_##trace_level##, _##keyword##, message, status);  \
+    }
+
+    void
+    ebpf_log_message_uint64(
+        ebpf_tracelog_level_t trace_level, ebpf_tracelog_keyword_t keyword, _In_z_ const char* message, uint64_t value);
+#define EBPF_LOG_MESSAGE_UINT64(trace_level, keyword, message, value)                \
+    if (TraceLoggingProviderEnabled(ebpf_tracelog_provider, trace_level, keyword)) { \
+        ebpf_log_message_uint64(_##trace_level##, _##keyword##, message, value);     \
+    }
+
+    void
+    ebpf_log_message_uint64_uint64(
+        ebpf_tracelog_level_t trace_level,
+        ebpf_tracelog_keyword_t keyword,
+        _In_z_ const char* message,
+        uint64_t value1,
+        uint64_t value2);
+#define EBPF_LOG_MESSAGE_UINT64_UINT64(trace_level, keyword, message, value1, value2)            \
+    if (TraceLoggingProviderEnabled(ebpf_tracelog_provider, trace_level, keyword)) {             \
+        ebpf_log_message_uint64_uint64(_##trace_level##, _##keyword##, message, value1, value2); \
+    }
+
+    void
+    ebpf_log_message_error(
+        ebpf_tracelog_level_t trace_level,
+        ebpf_tracelog_keyword_t keyword,
+        _In_z_ const char* message,
+        _In_ ebpf_result_t error);
+#define EBPF_LOG_MESSAGE_ERROR(trace_level, keyword, message, error)                 \
+    if (TraceLoggingProviderEnabled(ebpf_tracelog_provider, trace_level, keyword)) { \
+        ebpf_log_message_error(_##trace_level##, _##keyword##, message, error);      \
+    }
+
+    void
+    ebpf_log_message_wstring(
+        ebpf_tracelog_level_t trace_level,
+        ebpf_tracelog_keyword_t keyword,
+        _In_z_ const char* message,
+        _In_z_ const wchar_t* wstring);
+#define EBPF_LOG_MESSAGE_WSTRING(trace_level, keyword, message, wstring)             \
+    if (TraceLoggingProviderEnabled(ebpf_tracelog_provider, trace_level, keyword)) { \
+        ebpf_log_message_wstring(_##trace_level##, _##keyword##, message, wstring);  \
+    }
+
+    void
+    ebpf_log_message_guid_guid_string(
+        ebpf_tracelog_level_t trace_level,
+        ebpf_tracelog_keyword_t keyword,
+        _In_z_ const char* message,
+        _In_z_ const char* string,
+        _In_ const GUID guid1,
+        _In_ const GUID guid2);
```

Comment:
no "In" on value types, only pointer types.  Also value types need not be const
```suggestion
        GUID guid1,
        GUID guid2);
```
My understanding is that GUID is a struct.  If it's a pointer, let me know.

Reply:
Agree, inherited from previous implementation.

Reply:
Changed the existing implementation to use pointers instead.